src/WebJobs.Script.WebHost/Diagnostics/DeferredLogger.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Channels;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    /// <summary>
+    /// A logger that defers log entries to a channel.
+    /// </summary>
+    public class DeferredLogger : ILogger
+    {
+        private readonly Channel<DeferredLogEntry> _channel;
+        private readonly string _categoryName;
+        private IExternalScopeProvider _scopeProvider;
+
+        public DeferredLogger(Channel<DeferredLogEntry> channel, string categoryName, IExternalScopeProvider scopeProvider)
+        {
+            _channel = channel;
+            _categoryName = categoryName;
+            _scopeProvider = scopeProvider;
+        }
+
+        public IDisposable BeginScope<TState>(TState state) => _scopeProvider.Push(state);
+
+        public bool IsEnabled(LogLevel logLevel) => logLevel >= LogLevel.Debug;
```

Comment:
I don't think we want to hardcode suppressing logs at the debug level - let logging configuration do that.

```suggestion
        public bool IsEnabled(LogLevel logLevel) => logLevel != LogLevel.None;
```

Reply:
+1

Reply:
We want to make sure placeholder logs are not captured. That flag could be evaluated to determine enablement.

Reply:
I am restricting this to `Error` for now.